_episodes/01-raster-structure.md

```diff
@@ -15,10 +15,474 @@ objectives:
 - "To plot raster data with the `earthpy` package, we need to read in the image as a numpy array."
-- "`rasterio` stores CRS information in the PROJ4 dict format."
-- "Be careful when dealing with missing or bad data values."
+- "`rasterio` stores CRS information as a CRS object that can be converted to an EPSG code or PROJ4 string."
+- "The GeoTIFF file may or may not store the correct no data value(s). We can find the correct value(s) in the raster's external metadata or by plotting the raster."
 ---
-FIXME
 
-{% include links.md %}
+> ## Things You'll Need To Complete This Episode
+>
+> See the [lesson homepage]({{ site.baseurl }}) for detailed information about the software,
+> data, and other prerequisites you will need to work through the examples in this episode.
+{: .prereq}
 
+In this episode, we will introduce the fundamental principles, packages and
+metadata/raster attributes that are needed to work with raster data in Python. We will
+discuss some of the core metadata elements that we need to understand to work with
+rasters, including Coordinate Reference Systems, no data values, and resolution. We will also explore missing and bad
+data values as stored in a raster and how Python handles these elements.
+
+We will use three packages in this episode to work with raster data -
+`rasterio` for reading and writing rasters, `scipy` for descriptive statistics, and `earthpy.plot` for plotting. Make sure that you have these packages installed and imported.
+
+
+~~~
+import rasterio
+import earthpy.plot
+~~~
+{: .language-python}
+
+> ## Introduce the Data (TODO update link)
+>
+> A brief introduction to the datasets can be found on the 
+> [Geospatial workshop homepage](https://datacarpentry.org/geospatial-workshop/#data).
+> 
+> For more detailed information about the datasets, check
+out the [Geospatial workshop data
+page](http://datacarpentry.org/geospatial-workshop/data/).
+{: .callout}
+
+## View Raster File Attributes
+
+We will be working with a series of GeoTIFF files in this lesson. The
+GeoTIFF format contains a set of embedded tags with metadata about the raster
+data. We can use the function `rasterio.open()` to get metadata about our raster
+data before we read the whole file into Python. This let's you get a quick look at
+the shape and attributes of your data.
+
+
+~~~
+surface_model_HARV = rasterio.open("data/NEON-DS-Airborne-Remote-Sensing/HARV/DSM/HARV_dsmCrop.tif")
+surface_model_HARV.meta
+~~~
+{: .language-python}
+
+
+
+~~~
+{'driver': 'GTiff',
+ 'dtype': 'float64',
+ 'nodata': -9999.0,
+ 'width': 1697,
+ 'height': 1367,
+ 'count': 1,
+ 'crs': CRS.from_epsg(32618),
+ 'transform': Affine(1.0, 0.0, 731453.0,
+        0.0, -1.0, 4713838.0)}
+~~~
+{: .output}
+
+The first call to `rasterio.open()` opens the file and returns an object that we store in a variable, `surface_model_HARV`.
+This object has an attribute, `.meta`, which contains the metadata for the file we opened.
+
+
```

Comment:
Thanks, good suggestion.

The transform encodes how pixel coordinates are mapped to spatial coordinates, so it contains a matrix representation of a spatial projection that allows for projections that don't have EPSG codes. R reports the PROJ4 string if I remember correctly, this PROJ4 string is also accessible from the CRS object and we examine the PROJ4 string representation later in the lesson. I think rasterio chose to use the EPSG code as the representation of projections in the .meta dictionary instead of a proj4 string because some people prefer to work directly with EPSG codes.

Reply:
I included a short description of each metadata value, including the transform